python/tvm/relay/tensorrt.py

```diff
@@ -439,25 +445,47 @@ def reshape_whitelist_fn(attrs, args):  # pylint: disable=unused-variable
             print("reshape: new shape dims must be explicit.")
             return False
         if use_implicit_batch:
-            shape = list(map(int, args[0].checked_type.shape))
-            new_shape = list(map(int, attrs.newshape))
+            shape = args[0].checked_type.shape
+            new_shape = attrs.newshape
             if len(new_shape) == 0 or len(shape) == 0:
                 print("reshape: Can't reshape to or from scalar.")
                 return False
-            # TRT cannot modify batch dimension.
-            original_volume = np.prod(shape)
-            # First, resolve 0.
-            for i, value in enumerate(new_shape):
-                if value == 0:
-                    new_shape[i] = shape[i]
-            # Resolve -1.
-            for i, value in enumerate(new_shape):
-                if value == -1:
-                    new_shape[i] = original_volume // np.prod([x for x in new_shape if x != -1])
-            # Remove batch dimension and see if volumes match
-            if shape[0] != new_shape[0]:
-                print("reshape: can't modify batch dimension.")
-                return False
+
+            dynamic_reshape = False
```

Comment:
This can be reduced to `dynamic_reshape = any([isinstance(x, tvm.tir.expr.Any) for x in shape])`

Reply:
Done